components/FutureGraph.tsx

```diff
@@ -0,0 +1,318 @@
+    if (data) {
+      let currIndex = getCurrentTimeForecastIndex() + 1;
+
+      while (currIndex < data.forecast_values.length) {
+        const currentExpectedGenerationKW =
+          data.forecast_values[currIndex].expected_generation_kw;
+        const previousExpectedGenerationKW =
+          data.forecast_values[currIndex - 1].expected_generation_kw;
+        const formattedDate = formatter.format(
+          new Date(data.forecast_values[currIndex].target_datetime_utc)
+        );
+
+        if (currIndex === data.forecast_values.length - 1) {
+          if (currentExpectedGenerationKW > previousExpectedGenerationKW) {
+            return solarIncreasingText(formattedDate);
+          } else if (
+            currentExpectedGenerationKW < previousExpectedGenerationKW
+          ) {
+            return solarDecreasingText(formattedDate);
+          }
+        } else {
+          const nextExpectedGenerationKW =
+            data.forecast_values[currIndex + 1].expected_generation_kw;
+
+          if (
+            currentExpectedGenerationKW > previousExpectedGenerationKW &&
+            currentExpectedGenerationKW > nextExpectedGenerationKW
+          ) {
+            return solarIncreasingText(formattedDate);
+          } else if (
+            currentExpectedGenerationKW < previousExpectedGenerationKW &&
+            currentExpectedGenerationKW < nextExpectedGenerationKW
+          ) {
+            return solarDecreasingText(formattedDate);
+          }
+        }
+        currIndex += 1;
+      }
+    }
+
+    return '';
+  };
+
+  return (
+    <div className="relative my-2 w-full h-[260px] bg-ocf-gray-1000 rounded-2xl content-center">
+      <div className="flex flex-col w-full justify-start">
+        <div className="flex justify-end mt-[20px] mr-10 text-sm">
+          <FutureThresholdLegendIcon />
+        </div>
+        <ResponsiveContainer className="mt-[15px] " width="100%" height={100}>
+          <AreaChart
+            data={data?.forecast_values}
+            margin={{
+              top: 0,
+              right: 40,
+              left: 0,
+              bottom: 10,
+            }}
+          >
+            <defs>{generateGraphGradient()}</defs>
+            <YAxis
+              type="number"
+              domain={[0, 1.1]}
+              axisLine={false}
+              tick={false}
+            />
+            <Area
+              type="monotone"
+              dataKey="expected_generation_kw"
+              strokeWidth={2}
+              stroke="white"
+              strokeDasharray="2"
+              fill="url(#colorUv)"
+            >
+              <LabelList
+                dataKey="expected_generation_kw"
+                content={renderGraphLabels}
+              />
+            </Area>
+            <ReferenceLine
+              y={graphThreshold}
+              strokeWidth={2}
+              stroke="#FFD053"
+              strokeDasharray="2"
+            ></ReferenceLine>
```

Comment:
Can this self close?
```suggestion
            />
```